### PR TITLE
Add Deepkit

### PR DESCRIPTION
**Pull request by M-jerez**
Hi @moltar,  First of all thanks for this repo, I'm sure is becoming a reference with everything validation related in JS/Typescript! 👍

This PR adds the Deepkit validation library `@deepkit/types` to the benchmarks.  https://deepkit.io/

Any questions or amendments please let me know!

We should also close #895

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +27,9 @@
   "dependencies": {
     "@ailabs/ts-utils": "1.4.0",
     "@badrap/valita": "0.3.0",
+    "@deepkit/core": "^1.0.1-alpha.114",
+    "@deepkit/type": "^1.0.1-alpha.117",
+    "@deepkit/type-compiler": "^1.0.1-alpha.117",
```
Comment (moltar) [inline package.json:32]: Please pin deps to exact versions.

**Review by moltar — CHANGES_REQUESTED** [review]

**Comment by moltar** [comment]
Also, please run a linter `npm run lint`. Thanks!

**Comment by M-jerez** [comment]
@moltar all fixes applied, linting and testing passing in local.
I also did review the test so it better reflect deepkit's capabilities. 

Think everything should be ready to go 👍

**Comment by M-jerez** [comment]
@moltar  quick question about these conflicts with the test results `docs/results/node-20.json`. 
Should I remove my changes so there are no more conflicts (I guess there is an action to run the benchmark in this case) or should i keep overriding the result from main with mine to include the deepkit results?

**Comment by moltar** [comment]
> @moltar quick question about these conflicts with the test results `docs/results/node-20.json`. Should I remove my changes so there are no more conflicts (I guess there is an action to run the benchmark in this case) or should i keep overriding the result from main with mine to include the deepkit results?

Hey, yes, the tests will be auto-generated by CI, so it is safe to remove the file and that would fix the conflict.

**Review by moltar — APPROVED** [review]

**Pull request merged**